### ISSUE TITLE
How to obtain the result on Nuscenes test set?

### ISSUE THREAD

**Issue by czyczyyzc** [exiawsh/StreamPETR#34]
I trained the model with DD3D pre-trained weights, and obtained the results of 51.1 mAP and 59.5 NDS on Nuscenes validation set. And I want to further reproduce the results on Nuscenes test set, how can I achieve this?  Thanks very much.

It seems that there is no label on test set, and simply replace 'nuscenes2d_temporal_infos_val.pkl' with 'nuscenes2d_temporal_infos_test.pkl' in config file doesn't work.

**Comment by exiawsh** [comment]
Hi, thanks for your attention.
You should modify the config
1. change the training schedule 
num_iters_per_epoch = 34149 // (num_gpus * batch_size)
num_epochs = 60

2. change the image resolution to 1600x640
ida_aug_conf = {
        "resize_lim": (0.94, 1.25),
        "final_dim": (640, 1600),
        "bot_pct_lim": (0.0, 0.0),
        "rot_lim": (0.0, 0.0),
        "H": 900,
        "W": 1600,
        "rand_flip": True,
    } 

3. use data both on train and val set
ann_file=[data_root + 'nuscenes2d_temporal_infos_train.pkl', data_root + 'nuscenes2d_temporal_infos_val.pkl'],

**Comment by czyczyyzc** [comment]
Thank you very much. I will try this setting.

**Comment by jjlinghu** [comment]
> Thank you very much. I will try this setting.

Have you tried it? I also want to reproduce the results on Nuscenes test set. Is it possible to obtain the corresponding result by modifying as the author says? Thank you!

**Comment by czyczyyzc** [comment]
> > Thank you very much. I will try this setting.
> 
> Have you tried it? I also want to reproduce the results on Nuscenes test set. Is it possible to obtain the corresponding result by modifying as the author says? Thank you!

I have tried the setting the author provided, and I can reproduce the results on Nuscenes test set. My reproduced result is as follows:

"result": {"mAP": 0.565516945171988, "mATE": 0.47988312057653426, "mASE": 0.24136540290180744, "mAOE": 0.3226988221618697, "mAVE": 0.26263842280402855, "mAAE": 0.12512735965687488, "NDS": 0.6395871597758825}}

**Comment by jjlinghu** [comment]
> > > Thank you very much. I will try this setting.
> > 
> > 
> > Have you tried it? I also want to reproduce the results on Nuscenes test set. Is it possible to obtain the corresponding result by modifying as the author says? Thank you!
> 
> I have tried the setting the author provided, and I can reproduce the results on Nuscenes test set. My reproduced result is as follows:
> 
> "result": {"mAP": 0.565516945171988, "mATE": 0.47988312057653426, "mASE": 0.24136540290180744, "mAOE": 0.3226988221618697, "mAVE": 0.26263842280402855, "mAAE": 0.12512735965687488, "NDS": 0.6395871597758825}}

Thank you again!

**Comment by dodeokwangyooo** [comment]
> [@czyczyyzc](https://github.com/czyczyyzc) Hi, thanks for your attention. You should modify the config
> 
> 1. change the training schedule
>    num_iters_per_epoch = 34149 // (num_gpus * batch_size)
>    num_epochs = 60
> 2. change the image resolution to 1600x640
>    ida_aug_conf = {
>    "resize_lim": (0.94, 1.25),
>    "final_dim": (640, 1600),
>    "bot_pct_lim": (0.0, 0.0),
>    "rot_lim": (0.0, 0.0),
>    "H": 900,
>    "W": 1600,
>    "rand_flip": True,
>    }
> 3. use data both on train and val set
>    ann_file=[data_root + 'nuscenes2d_temporal_infos_train.pkl', data_root + 'nuscenes2d_temporal_infos_val.pkl'],

I want to use it in my own mmdet program but I get an error, please ask which part of this code is related to your program to modify?
”ann_file=[data_root + 'nuscenes2d_temporal_infos_train.pkl'， data_root + 'nuscenes2d_temporal_infos_val.pkl'] “
